migrations/2023-07-11-084714_receive_activity_table/up.sql

```diff
@@ -0,0 +1,23 @@
+create table sent_activity (
+    ap_id text primary key,
+    data jsonb not null,
+    sensitive boolean not null,
+    published timestamp not null default now()
+);
+
+create table received_activity (
+    ap_id text primary key,
+    published timestamp not null default now()
+);
+
+insert into sent_activity(ap_id, data, sensitive, published)
+    select ap_id, data, sensitive, published
+    from activity
+    where local = true;
+
+insert into received_activity(ap_id, published)
+    select ap_id, published
+    from activity
+    where local = false;
+
+drop table activity;
```

Comment:
since the table has probably grown very large for many instances, this is a pretty expensive migration - the table might have 100 million + rows for people that haven't manually cleaned it. So this would cause some significant downtime (hard to estimate).

It might make sense to add a filter to the copies as `where local = x AND id > (select max(id) from activity) - 100000` so only the most recent 100k activities are considered.

The reason my suggestion uses ID and not published is because there is no index on published . on a related note, there probably should be an index on published.